<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction â€¢ biomeUtils</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="biomeUtils">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomeUtils</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.06</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Introduction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction</h1>
            
      
      
      <div class="hidden name"><code>Introduction.Rmd</code></div>

    </div>

    
    
<div id="biomeutils" class="section level1">
<h1 class="hasAnchor">
<a href="#biomeutils" class="anchor"></a>biomeUtils</h1>
<p><code>biomeUtils</code> is part of the RIVM-ToolBox project aimed at providing standard set of tools that interact with open tools for a wide array of data analytics, including microbiomics. The RIVM-ToolBox is a set of individual R tools focused towards different goals/functionalities.</p>
<ul>
<li><p>Data handling<br>
Outputs for standard data generating pipelines/workflows.</p></li>
<li><p>Data analytics<br>
Common data analytics including basic statistics.</p></li>
<li><p>Data visualization<br>
Data visualization of different data types.</p></li>
</ul>
<div id="setup" class="section level2">
<h2 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h2>
<div id="install" class="section level3">
<h3 class="hasAnchor">
<a href="#install" class="anchor"></a>Install</h3>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"microsud/biomUtils"</span><span class="op">)</span></pre></div>
</div>
<div id="load-pkg" class="section level3">
<h3 class="hasAnchor">
<a href="#load-pkg" class="anchor"></a>Load pkg</h3>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: phyloseq</span>
<span class="co">#&gt; Loading required package: microbiome</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; microbiome R package (microbiome.github.com)</span>
<span class="co">#&gt;     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Copyright (C) 2011-2020 Leo Lahti, </span>
<span class="co">#&gt;     Sudarshan Shetty et al. &lt;microbiome.github.io&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'microbiome'</span>
<span class="co">#&gt; The following object is masked from 'package:ggplot2':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     alpha</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     transform</span></pre></div>
</div>
</div>
<div id="data-handling" class="section level2">
<h2 class="hasAnchor">
<a href="#data-handling" class="anchor"></a>Data handling</h2>
<div id="data" class="section level3">
<h3 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h3>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">FuentesIliGutData</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 905 taxa and 589 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 589 samples by 61 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 905 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 905 tips and 904 internal nodes ]</span></pre></div>
</div>
<div id="get-tibbles" class="section level3">
<h3 class="hasAnchor">
<a href="#get-tibbles" class="anchor"></a>Get tibbles</h3>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="co"># get otu table in tibble format</span>
<span class="va">otu_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTibble.html">getAbundanceTibble</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                              select_rows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ASV302"</span>, <span class="st">"ASV82"</span>, <span class="st">"ASV410"</span>, <span class="st">"ASV2332"</span><span class="op">)</span>,
                              select_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample_1"</span>, <span class="st">"sample_5"</span>, <span class="st">"sample_6"</span><span class="op">)</span>,
                              column_id <span class="op">=</span> <span class="st">"FeatureID"</span><span class="op">)</span>

<span class="co"># get taxa table in tibble format</span>
<span class="va">tax_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTibble.html">getTaxaTibble</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                         select_rows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ASV302"</span>, <span class="st">"ASV82"</span>, <span class="st">"ASV410"</span>, <span class="st">"ASV2332"</span><span class="op">)</span>,
                         select_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Phylum"</span>, <span class="st">"Genus"</span><span class="op">)</span>,
                         column_id <span class="op">=</span> <span class="st">"FeatureID"</span><span class="op">)</span>

<span class="co"># get sample data in tibble format</span>
<span class="va">meta_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTibble.html">getSampleTibble</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                            select_rows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample_1"</span>, <span class="st">"sample_5"</span>, <span class="st">"sample_6"</span><span class="op">)</span>,
                            select_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"participant_id"</span>, <span class="st">"ILI"</span>, <span class="st">"age"</span><span class="op">)</span>,
                            column_id <span class="op">=</span> <span class="st">"FeatureID"</span><span class="op">)</span></pre></div>
</div>
<div id="subsetfilter-phyloseq" class="section level3">
<h3 class="hasAnchor">
<a href="#subsetfilter-phyloseq" class="anchor"></a>Subset/Filter phyloseq</h3>
<div class="sourceCode" id="cb5"><pre class="downlit">

<span class="co"># Filter by phyloseq like subset_*</span>
<span class="va">ps.filtered.samples</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                         <span class="va">ILI</span> <span class="op">==</span> <span class="st">"C"</span> <span class="op">&amp;</span> <span class="va">BMI</span> <span class="op">&lt;</span> <span class="fl">26</span><span class="op">)</span>
<span class="va">ps.filtered.samples</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 901 taxa and 91 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 91 samples by 61 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 901 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 901 tips and 900 internal nodes ]</span>

<span class="va">ps.filtered.taxa</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterTaxaData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                    <span class="va">Phylum</span><span class="op">==</span><span class="st">"Firmicutes"</span><span class="op">)</span>

<span class="va">ps.filtered.taxa</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 758 taxa and 589 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 589 samples by 61 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 758 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 758 tips and 757 internal nodes ]</span>

<span class="co"># Filter by names like prune_*</span>
<span class="va">sams.select</span> <span class="op">&lt;-</span> <span class="fu">sample_names</span><span class="op">(</span><span class="va">FuentesIliGutData</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="va">ps.filter.by.sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleByNames</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                               ids <span class="op">=</span> <span class="va">sams.select</span>,
                                               keep <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">tax.select</span> <span class="op">&lt;-</span> <span class="fu">taxa_names</span><span class="op">(</span><span class="va">FuentesIliGutData</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="va">ps.filter.by.tax.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterTaxaByNames</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                           ids <span class="op">=</span> <span class="va">tax.select</span>,
                                           keep <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
</div>
<div id="compare-phyloseqs" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-phyloseqs" class="anchor"></a>Compare phyloseqs</h3>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">ps1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span>

<span class="va">ps2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"L1"</span><span class="op">)</span>

<span class="va">ps3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"L2"</span><span class="op">)</span>


<span class="va">ps.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"C"</span> <span class="op">=</span> <span class="va">ps1</span>, <span class="st">"L1"</span> <span class="op">=</span> <span class="va">ps2</span>, <span class="st">"L2"</span> <span class="op">=</span> <span class="va">ps3</span><span class="op">)</span>

<span class="fu"><a href="../reference/comparePhyloseq.html">comparePhyloseq</a></span><span class="op">(</span><span class="va">ps.list</span><span class="op">)</span>
<span class="co">#&gt;   input ntaxa nsample min.reads max.reads total.read average.reads singletons</span>
<span class="co">#&gt; 1     C   901     184     10975    143946    7658743      41623.60          0</span>
<span class="co">#&gt; 2    L1   902     213     12715    140779    8568244      40226.50          0</span>
<span class="co">#&gt; 3    L2   896     192     24868    194135   15127300      78788.02          0</span>
<span class="co">#&gt;    sparsity</span>
<span class="co">#&gt; 1 0.6717355</span>
<span class="co">#&gt; 2 0.6678794</span>
<span class="co">#&gt; 3 0.6372652</span></pre></div>
</div>
</div>
<div id="calculations" class="section level2">
<h2 class="hasAnchor">
<a href="#calculations" class="anchor"></a>Calculations</h2>
<div id="phylogenetic-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#phylogenetic-diversity" class="anchor"></a>Phylogenetic diversity</h3>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="co"># reduce size for example</span>
<span class="va">ps1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span>

<span class="va">meta_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculatePD.html">calculatePD</a></span><span class="op">(</span><span class="va">ps1</span>, justDF<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># check</span>
<span class="va">meta_tib</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"PD"</span>, <span class="st">"SR"</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt;                 PD  SR</span>
<span class="co">#&gt; sample_1  967.3003 259</span>
<span class="co">#&gt; sample_2 1035.1305 291</span>
<span class="co">#&gt; sample_3 1189.0248 336</span></pre></div>
</div>
<div id="prevalence" class="section level3">
<h3 class="hasAnchor">
<a href="#prevalence" class="anchor"></a>Prevalence</h3>
<div class="sourceCode" id="cb8"><pre class="downlit">

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">prev_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                          return_rank<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Family"</span>, <span class="st">"Genus"</span><span class="op">)</span>,
                          return_taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ASV4"</span>, <span class="st">"ASV17"</span> , <span class="st">"ASV85"</span>, <span class="st">"ASV83"</span><span class="op">)</span>,
                          sort<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "Taxa"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">prev_tib</span><span class="op">)</span>
<span class="co">#&gt;    Taxa prevalence           Family         Genus</span>
<span class="co">#&gt; 1  ASV4  0.9881154  Lachnospiraceae       Blautia</span>
<span class="co">#&gt; 2 ASV17  0.9847199 Streptococcaceae Streptococcus</span>
<span class="co">#&gt; 3 ASV85  0.7758913  Lachnospiraceae     Roseburia</span>
<span class="co">#&gt; 4 ASV83  0.7045840   Bacteroidaceae   Bacteroides</span></pre></div>
</div>
<div id="fold-difference" class="section level3">
<h3 class="hasAnchor">
<a href="#fold-difference" class="anchor"></a>Fold Difference</h3>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="co"># Keep only two groups</span>
<span class="va">ps1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">!=</span> <span class="st">"L2"</span><span class="op">)</span>

<span class="va">taxa_fd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateTaxaFoldDifference.html">calculateTaxaFoldDifference</a></span><span class="op">(</span><span class="va">ps1</span>, group<span class="op">=</span><span class="st">"ILI"</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "Taxa"</span>
<span class="co">#&gt; Joining, by = "Taxa"</span>
<span class="co">#&gt; Joining, by = c("Taxa", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")</span>
<span class="co"># check</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">taxa_fd</span><span class="op">)</span>
<span class="co">#&gt;     Domain     Phylum      Class         Order          Family</span>
<span class="co">#&gt; 1 Bacteria Firmicutes Clostridia Clostridiales Lachnospiraceae</span>
<span class="co">#&gt; 2 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt; 3 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt; 4 Bacteria Firmicutes Clostridia Clostridiales Lachnospiraceae</span>
<span class="co">#&gt; 5 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt; 6 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt;                          Genus        Species FoldDifference FeatureID</span>
<span class="co">#&gt; 1 Lachnospiraceae NC2004 group           &lt;NA&gt;   -0.167551359    ASV302</span>
<span class="co">#&gt; 2      Ruminococcaceae UCG-013           &lt;NA&gt;   -0.100108612    ASV636</span>
<span class="co">#&gt; 3                         &lt;NA&gt;           &lt;NA&gt;    0.157777020    ASV500</span>
<span class="co">#&gt; 4             Fusicatenibacter saccharivorans   -0.124023091      ASV7</span>
<span class="co">#&gt; 5                         &lt;NA&gt;           &lt;NA&gt;   -0.068360663   ASV2617</span>
<span class="co">#&gt; 6                         &lt;NA&gt;           &lt;NA&gt;   -0.002698838    ASV148</span>
<span class="co">#&gt;   Prevalence.C Prevalence.L1 Enriched</span>
<span class="co">#&gt; 1    0.8478261     0.8309859        C</span>
<span class="co">#&gt; 2    0.2228261     0.1830986        C</span>
<span class="co">#&gt; 3    0.2989130     0.4178404       L1</span>
<span class="co">#&gt; 4    0.9945652     0.9624413        C</span>
<span class="co">#&gt; 5    0.2336957     0.1784038        C</span>
<span class="co">#&gt; 6    0.3206522     0.3192488        C</span></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 10 x64 (build 19042)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   </span>
<span class="co">#&gt; [3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        </span>
<span class="co">#&gt; [5] LC_TIME=English_Netherlands.1252    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] biomeUtils_0.0.06 microbiome_1.11.2 ggplot2_3.3.2     phyloseq_1.33.0  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] Biobase_2.50.0      tidyr_1.1.2         bit64_4.0.5        </span>
<span class="co">#&gt;  [4] jsonlite_1.7.1      splines_4.0.3       foreach_1.5.1      </span>
<span class="co">#&gt;  [7] assertthat_0.2.1    stats4_4.0.3        blob_1.2.1         </span>
<span class="co">#&gt; [10] yaml_2.2.1          pillar_1.4.6        RSQLite_2.2.1      </span>
<span class="co">#&gt; [13] backports_1.2.0     lattice_0.20-41     quadprog_1.5-8     </span>
<span class="co">#&gt; [16] phangorn_2.5.5      glue_1.4.2          digest_0.6.27      </span>
<span class="co">#&gt; [19] XVector_0.30.0      picante_1.8.2       colorspace_1.4-1   </span>
<span class="co">#&gt; [22] htmltools_0.5.1.1   Matrix_1.2-18       plyr_1.8.6         </span>
<span class="co">#&gt; [25] pkgconfig_2.0.3     zlibbioc_1.36.0     purrr_0.3.4        </span>
<span class="co">#&gt; [28] scales_1.1.1        Rtsne_0.15          tibble_3.0.4       </span>
<span class="co">#&gt; [31] mgcv_1.8-33         generics_0.1.0      IRanges_2.24.0     </span>
<span class="co">#&gt; [34] ellipsis_0.3.1      withr_2.3.0         BiocGenerics_0.36.0</span>
<span class="co">#&gt; [37] survival_3.2-7      magrittr_1.5        crayon_1.3.4       </span>
<span class="co">#&gt; [40] memoise_1.1.0       evaluate_0.14       fs_1.5.0           </span>
<span class="co">#&gt; [43] nlme_3.1-149        MASS_7.3-53         vegan_2.5-6        </span>
<span class="co">#&gt; [46] textshaping_0.1.2   tools_4.0.3         data.table_1.13.0  </span>
<span class="co">#&gt; [49] lifecycle_0.2.0     stringr_1.4.0       Rhdf5lib_1.12.0    </span>
<span class="co">#&gt; [52] S4Vectors_0.28.0    munsell_0.5.0       cluster_2.1.0      </span>
<span class="co">#&gt; [55] Biostrings_2.58.0   ade4_1.7-15         compiler_4.0.3     </span>
<span class="co">#&gt; [58] pkgdown_1.6.1       systemfonts_0.3.2   rlang_0.4.8        </span>
<span class="co">#&gt; [61] rhdf5_2.34.0        grid_4.0.3          iterators_1.0.13   </span>
<span class="co">#&gt; [64] rhdf5filters_1.2.0  biomformat_1.17.0   rstudioapi_0.11    </span>
<span class="co">#&gt; [67] igraph_1.2.6        rmarkdown_2.4       gtable_0.3.0       </span>
<span class="co">#&gt; [70] codetools_0.2-16    multtest_2.45.0     DBI_1.1.0          </span>
<span class="co">#&gt; [73] reshape2_1.4.4      R6_2.5.0            knitr_1.30         </span>
<span class="co">#&gt; [76] dplyr_1.0.2         bit_4.0.4           fastmatch_1.1-0    </span>
<span class="co">#&gt; [79] rprojroot_1.3-2     ragg_0.4.0          permute_0.9-5      </span>
<span class="co">#&gt; [82] desc_1.2.0          ape_5.4-1           stringi_1.5.3      </span>
<span class="co">#&gt; [85] parallel_4.0.3      Rcpp_1.0.5          DECIPHER_2.18.1    </span>
<span class="co">#&gt; [88] vctrs_0.3.4         tidyselect_1.1.0    xfun_0.17</span></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sudarshan Shetty, Person Person, Person Person.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
