<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction â€¢ biomeUtils</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="biomeUtils">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biomeUtils</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.09</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Introduction.html">Introduction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/RIVM-IIV-Microbiome/biomeUtils/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="Introduction_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/RIVM-IIV-Microbiome/biomeUtils/blob/master/vignettes/Introduction.Rmd"><code>vignettes/Introduction.Rmd</code></a></small>
      <div class="hidden name"><code>Introduction.Rmd</code></div>

    </div>

    
    
<div id="biomeutils" class="section level1">
<h1 class="hasAnchor">
<a href="#biomeutils" class="anchor"></a>biomeUtils</h1>
<p><code>biomeUtils</code> is part of the RIVM-ToolBox project aimed at providing standard set of tools that interact with open tools for a wide array of data analytics, including microbiomics. The RIVM-ToolBox is a set of individual R tools focused towards different goals/functionalities.</p>
<ul>
<li><p><code>biomeUtils</code> : Data handling<br>
Outputs for standard data generating pipelines/workflows.</p></li>
<li><p><code>biomeStats</code>: Data analytics<br>
Common data analytics including basic statistics.</p></li>
<li><p><code>biomeViz</code>: Data visualization<br>
Data visualization of different data types.</p></li>
</ul>
<div id="setup" class="section level2">
<h2 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h2>
<div id="install" class="section level3">
<h3 class="hasAnchor">
<a href="#install" class="anchor"></a>Install</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span><span class="op">(</span><span class="st">"microsud/biomUtils"</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-pkg" class="section level3">
<h3 class="hasAnchor">
<a href="#load-pkg" class="anchor"></a>Load pkg</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="co">#&gt; Loading required package: phyloseq</span>
<span class="co">#&gt; Loading required package: microbiome</span>
<span class="co">#&gt; Loading required package: ggplot2</span>
<span class="co">#&gt; Warning: package 'ggplot2' was built under R version 4.0.5</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; microbiome R package (microbiome.github.com)</span>
<span class="co">#&gt;     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;  Copyright (C) 2011-2020 Leo Lahti, </span>
<span class="co">#&gt;     Sudarshan Shetty et al. &lt;microbiome.github.io&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'microbiome'</span>
<span class="co">#&gt; The following object is masked from 'package:ggplot2':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     alpha</span>
<span class="co">#&gt; The following object is masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     transform</span></code></pre></div>
</div>
</div>
<div id="data-handling" class="section level2">
<h2 class="hasAnchor">
<a href="#data-handling" class="anchor"></a>Data handling</h2>
<div id="data" class="section level3">
<h3 class="hasAnchor">
<a href="#data" class="anchor"></a>Data</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">FuentesIliGutData</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 905 taxa and 589 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 589 samples by 61 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 905 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 905 tips and 904 internal nodes ]</span></code></pre></div>
</div>
<div id="get-tibbles" class="section level3">
<h3 class="hasAnchor">
<a href="#get-tibbles" class="anchor"></a>Get tibbles</h3>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="co"># get otu table in tibble format</span>
<span class="va">otu_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTibble.html">getAbundanceTibble</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                              select_rows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ASV302"</span>, <span class="st">"ASV82"</span>, <span class="st">"ASV410"</span>, <span class="st">"ASV2332"</span><span class="op">)</span>,
                              select_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample_1"</span>, <span class="st">"sample_5"</span>, <span class="st">"sample_6"</span><span class="op">)</span>,
                              column_id <span class="op">=</span> <span class="st">"FeatureID"</span><span class="op">)</span>

<span class="co"># get taxa table in tibble format</span>
<span class="va">tax_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTibble.html">getTaxaTibble</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                         select_rows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ASV302"</span>, <span class="st">"ASV82"</span>, <span class="st">"ASV410"</span>, <span class="st">"ASV2332"</span><span class="op">)</span>,
                         select_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Phylum"</span>, <span class="st">"Genus"</span><span class="op">)</span>,
                         column_id <span class="op">=</span> <span class="st">"FeatureID"</span><span class="op">)</span>

<span class="co"># get sample data in tibble format</span>
<span class="va">meta_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getTibble.html">getSampleTibble</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                            select_rows <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"sample_1"</span>, <span class="st">"sample_5"</span>, <span class="st">"sample_6"</span><span class="op">)</span>,
                            select_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"participant_id"</span>, <span class="st">"ILI"</span>, <span class="st">"age"</span><span class="op">)</span>,
                            column_id <span class="op">=</span> <span class="st">"FeatureID"</span><span class="op">)</span></code></pre></div>
</div>
<div id="subsetfilter-phyloseq" class="section level3">
<h3 class="hasAnchor">
<a href="#subsetfilter-phyloseq" class="anchor"></a>Subset/Filter phyloseq</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># Filter by phyloseq like subset_*</span>
<span class="va">ps.filtered.samples</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                         <span class="va">ILI</span> <span class="op">==</span> <span class="st">"C"</span> <span class="op">&amp;</span> <span class="va">BMI</span> <span class="op">&lt;</span> <span class="fl">26</span><span class="op">)</span>
<span class="va">ps.filtered.samples</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 901 taxa and 91 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 91 samples by 61 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 901 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 901 tips and 900 internal nodes ]</span>

<span class="va">ps.filtered.taxa</span>  <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterTaxaData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                    <span class="va">Phylum</span><span class="op">==</span><span class="st">"Firmicutes"</span><span class="op">)</span>

<span class="va">ps.filtered.taxa</span>
<span class="co">#&gt; phyloseq-class experiment-level object</span>
<span class="co">#&gt; otu_table()   OTU Table:         [ 758 taxa and 589 samples ]</span>
<span class="co">#&gt; sample_data() Sample Data:       [ 589 samples by 61 sample variables ]</span>
<span class="co">#&gt; tax_table()   Taxonomy Table:    [ 758 taxa by 7 taxonomic ranks ]</span>
<span class="co">#&gt; phy_tree()    Phylogenetic Tree: [ 758 tips and 757 internal nodes ]</span>

<span class="co"># Filter by names like prune_*</span>
<span class="va">sams.select</span> <span class="op">&lt;-</span> <span class="fu">sample_names</span><span class="op">(</span><span class="va">FuentesIliGutData</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="va">ps.filter.by.sam.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleByNames</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                               ids <span class="op">=</span> <span class="va">sams.select</span>,
                                               keep <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="va">tax.select</span> <span class="op">&lt;-</span> <span class="fu">taxa_names</span><span class="op">(</span><span class="va">FuentesIliGutData</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>
<span class="va">ps.filter.by.tax.names</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterTaxaByNames</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                                           ids <span class="op">=</span> <span class="va">tax.select</span>,
                                           keep <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
</div>
<div id="check-for-polyphyly" class="section level3">
<h3 class="hasAnchor">
<a href="#check-for-polyphyly" class="anchor"></a>Check for polyphyly</h3>
<p>Check for polyphyletic taxa in <code>tax_table</code>. Useful to check this before aggregating at any level. Here, for e.g.Â <em>Eubacterium</em> is is in both <code>Lachnospiraceae</code> and <code>EUbacteriaceae</code> family.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: 'dplyr'</span>
<span class="co">#&gt; The following objects are masked from 'package:stats':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter, lag</span>
<span class="co">#&gt; The following objects are masked from 'package:base':</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     intersect, setdiff, setequal, union</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">polydf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/checkPolyphyletic.html">checkPolyphyletic</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                            taxa_level <span class="op">=</span> <span class="st">"Genus"</span>,
                            return_df <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="va">polydf</span>
<span class="co">#&gt; # A tibble: 6 x 7</span>
<span class="co">#&gt; # Groups:   Genus [3]</span>
<span class="co">#&gt;   Domain   Phylum     Class      Order         Family          Genus           n</span>
<span class="co">#&gt;   &lt;chr&gt;    &lt;chr&gt;      &lt;chr&gt;      &lt;chr&gt;         &lt;chr&gt;           &lt;chr&gt;       &lt;int&gt;</span>
<span class="co">#&gt; 1 Bacteria Firmicutes Clostridia Clostridiales Lachnospiraceae Eubacterium     2</span>
<span class="co">#&gt; 2 Bacteria Firmicutes Clostridia Clostridiales Lachnospiraceae Clostridium     2</span>
<span class="co">#&gt; 3 Bacteria Firmicutes Clostridia Clostridiales Lachnospiraceae Ruminococc~     2</span>
<span class="co">#&gt; 4 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae Ruminococc~     2</span>
<span class="co">#&gt; 5 Bacteria Firmicutes Clostridia Clostridiales Eubacteriaceae  Eubacterium     2</span>
<span class="co">#&gt; 6 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae Clostridium     2</span></code></pre></div>
</div>
<div id="compare-phyloseqs" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-phyloseqs" class="anchor"></a>Compare phyloseqs</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">ps1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span>

<span class="va">ps2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"L1"</span><span class="op">)</span>

<span class="va">ps3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"L2"</span><span class="op">)</span>


<span class="va">ps.list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"C"</span> <span class="op">=</span> <span class="va">ps1</span>, <span class="st">"L1"</span> <span class="op">=</span> <span class="va">ps2</span>, <span class="st">"L2"</span> <span class="op">=</span> <span class="va">ps3</span><span class="op">)</span>

<span class="fu"><a href="../reference/comparePhyloseq.html">comparePhyloseq</a></span><span class="op">(</span><span class="va">ps.list</span><span class="op">)</span>
<span class="co">#&gt;   input ntaxa nsample min.reads max.reads total.read average.reads singletons</span>
<span class="co">#&gt; 1     C   901     184     10975    143946    7658743      41623.60          0</span>
<span class="co">#&gt; 2    L1   902     213     12715    140779    8568244      40226.50          0</span>
<span class="co">#&gt; 3    L2   896     192     24868    194135   15127300      78788.02          0</span>
<span class="co">#&gt;    sparsity</span>
<span class="co">#&gt; 1 0.6717355</span>
<span class="co">#&gt; 2 0.6678794</span>
<span class="co">#&gt; 3 0.6372652</span></code></pre></div>
</div>
</div>
<div id="calculations" class="section level2">
<h2 class="hasAnchor">
<a href="#calculations" class="anchor"></a>Calculations</h2>
<div id="phylogenetic-diversity" class="section level3">
<h3 class="hasAnchor">
<a href="#phylogenetic-diversity" class="anchor"></a>Phylogenetic diversity</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="co"># reduce size for example</span>
<span class="va">ps1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">==</span> <span class="st">"C"</span><span class="op">)</span>

<span class="va">meta_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculatePD.html">calculatePD</a></span><span class="op">(</span><span class="va">ps1</span>, justDF<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co"># check</span>
<span class="va">meta_tib</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"PD"</span>, <span class="st">"SR"</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt;                 PD  SR</span>
<span class="co">#&gt; sample_1  967.3003 259</span>
<span class="co">#&gt; sample_2 1035.1305 291</span>
<span class="co">#&gt; sample_3 1189.0248 336</span></code></pre></div>
</div>
<div id="prevalence" class="section level3">
<h3 class="hasAnchor">
<a href="#prevalence" class="anchor"></a>Prevalence</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">prev_tib</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getPrevalence.html">getPrevalence</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>,
                          return_rank<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Family"</span>, <span class="st">"Genus"</span><span class="op">)</span>,
                          return_taxa <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ASV4"</span>, <span class="st">"ASV17"</span> , <span class="st">"ASV85"</span>, <span class="st">"ASV83"</span><span class="op">)</span>,
                          sort<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "Taxa"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">prev_tib</span><span class="op">)</span>
<span class="co">#&gt;    Taxa prevalence           Family         Genus</span>
<span class="co">#&gt; 1  ASV4  0.9881154  Lachnospiraceae       Blautia</span>
<span class="co">#&gt; 2 ASV17  0.9847199 Streptococcaceae Streptococcus</span>
<span class="co">#&gt; 3 ASV85  0.7758913  Lachnospiraceae     Roseburia</span>
<span class="co">#&gt; 4 ASV83  0.7045840   Bacteroidaceae   Bacteroides</span></code></pre></div>
</div>
<div id="fold-difference" class="section level3">
<h3 class="hasAnchor">
<a href="#fold-difference" class="anchor"></a>Fold Difference</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="co"># Keep only two groups</span>
<span class="va">ps1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">!=</span> <span class="st">"L2"</span><span class="op">)</span>

<span class="va">taxa_fd</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/calculateTaxaFoldDifference.html">calculateTaxaFoldDifference</a></span><span class="op">(</span><span class="va">ps1</span>, group<span class="op">=</span><span class="st">"ILI"</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "Taxa"</span>
<span class="co">#&gt; Joining, by = "Taxa"</span>
<span class="co">#&gt; Joining, by = c("Taxa", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")</span>
<span class="co"># check</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">taxa_fd</span><span class="op">)</span>
<span class="co">#&gt;     Domain     Phylum      Class         Order          Family</span>
<span class="co">#&gt; 1 Bacteria Firmicutes Clostridia Clostridiales Lachnospiraceae</span>
<span class="co">#&gt; 2 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt; 3 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt; 4 Bacteria Firmicutes Clostridia Clostridiales Lachnospiraceae</span>
<span class="co">#&gt; 5 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt; 6 Bacteria Firmicutes Clostridia Clostridiales Ruminococcaceae</span>
<span class="co">#&gt;                          Genus        Species FoldDifference FeatureID</span>
<span class="co">#&gt; 1 Lachnospiraceae NC2004 group           &lt;NA&gt;   -0.167551359    ASV302</span>
<span class="co">#&gt; 2      Ruminococcaceae UCG-013           &lt;NA&gt;   -0.100108612    ASV636</span>
<span class="co">#&gt; 3                         &lt;NA&gt;           &lt;NA&gt;    0.157777020    ASV500</span>
<span class="co">#&gt; 4             Fusicatenibacter saccharivorans   -0.124023091      ASV7</span>
<span class="co">#&gt; 5                         &lt;NA&gt;           &lt;NA&gt;   -0.068360663   ASV2617</span>
<span class="co">#&gt; 6                         &lt;NA&gt;           &lt;NA&gt;   -0.002698838    ASV148</span>
<span class="co">#&gt;   Prevalence.C Prevalence.L1 Enriched</span>
<span class="co">#&gt; 1    0.8478261     0.8309859        C</span>
<span class="co">#&gt; 2    0.2228261     0.1830986        C</span>
<span class="co">#&gt; 3    0.2989130     0.4178404       L1</span>
<span class="co">#&gt; 4    0.9945652     0.9624413        C</span>
<span class="co">#&gt; 5    0.2336957     0.1784038        C</span>
<span class="co">#&gt; 6    0.3206522     0.3192488        C</span></code></pre></div>
</div>
<div id="melting" class="section level3">
<h3 class="hasAnchor">
<a href="#melting" class="anchor"></a>Melting</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">FuentesIliGutData</span>, <span class="va">ILI</span> <span class="op">!=</span> <span class="st">"L2"</span><span class="op">)</span>
<span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/rarefy_even_depth.html">rarefy_even_depth</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>
<span class="co">#&gt; You set `rngseed` to FALSE. Make sure you've set &amp; recorded</span>
<span class="co">#&gt;  the random seed of your session for reproducibility.</span>
<span class="co">#&gt; See `?set.seed`</span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; 1OTUs were removed because they are no longer </span>
<span class="co">#&gt; present in any sample after random subsampling</span>
<span class="co">#&gt; ...</span>
<span class="va">dist.mat</span> <span class="op">&lt;-</span> <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/distance.html">distance</a></span><span class="op">(</span><span class="va">ps</span>, <span class="st">"bray"</span><span class="op">)</span>
<span class="va">dist.melt.sample</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/meltDistanceToTable.html">meltDistanceToTable</a></span><span class="op">(</span><span class="va">ps</span>,
                                        dist_mat <span class="op">=</span> <span class="va">dist.mat</span>,
                                        name_dist_column <span class="op">=</span> <span class="st">"Bray-Curtis"</span>,
                                        select_cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"participant_id"</span>, <span class="st">"ILI"</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">dist.melt.sample</span><span class="op">)</span>
<span class="co">#&gt;         S1       S2 Bray.Curtis participant_id_S1 ILI_S1 participant_id_S2</span>
<span class="co">#&gt; 1 sample_2 sample_1   0.5588155              ELD2      C              ELD1</span>
<span class="co">#&gt; 2 sample_3 sample_1   0.5876082              ELD3      C              ELD1</span>
<span class="co">#&gt; 3 sample_4 sample_1   0.5488838              ELD4      C              ELD1</span>
<span class="co">#&gt; 4 sample_5 sample_1   0.6427335              ELD5     L1              ELD1</span>
<span class="co">#&gt; 5 sample_6 sample_1   0.7339408              ELD6      C              ELD1</span>
<span class="co">#&gt; 6 sample_7 sample_1   0.5805011              ELD7     L1              ELD1</span>
<span class="co">#&gt;   ILI_S2</span>
<span class="co">#&gt; 1      C</span>
<span class="co">#&gt; 2      C</span>
<span class="co">#&gt; 3      C</span>
<span class="co">#&gt; 4      C</span>
<span class="co">#&gt; 5      C</span>
<span class="co">#&gt; 6      C</span></code></pre></div>
</div>
</div>
</div>
<div id="pipeing-steps" class="section level1">
<h1 class="hasAnchor">
<a href="#pipeing-steps" class="anchor"></a>Pipeâ€™ingâ€™ steps</h1>
<p>Read <code><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></code> as <code>and then</code></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">biomeUtils</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span> <span class="co"># pipe function</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="st">"FuentesIliGutData"</span><span class="op">)</span>
<span class="va">FuentesIliGutData</span> <span class="op">%&gt;%</span> <span class="co"># Take FuentesIliGutData phyloseq object and then</span>
  <span class="fu"><a href="../reference/uniteGenusSpeciesNames.html">uniteGenusSpeciesNames</a></span><span class="op">(</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># join genus and species columns and then</span>
  <span class="fu"><a href="../reference/filterPhyloseq.html">filterSampleData</a></span><span class="op">(</span><span class="va">ILI</span> <span class="op">!=</span> <span class="st">"L1"</span> <span class="op">&amp;</span> <span class="va">BMI</span> <span class="op">&lt;</span> <span class="fl">30</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># Remove 'L1' group and keep those with BMI less than 30 and then</span>
  <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/transform.html">transform</a></span><span class="op">(</span><span class="va">.</span>, <span class="st">"compositional"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># convert to compositional and then</span>
  <span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/core.html">core</a></span><span class="op">(</span><span class="va">.</span>, <span class="fl">0.0001</span>, <span class="fl">0.01</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># select ASVs with 0.0001 in at least 1% samples and then</span>
  <span class="fu"><a href="../reference/calculateTaxaFoldDifference.html">calculateTaxaFoldDifference</a></span><span class="op">(</span>group<span class="op">=</span><span class="st">"ILI"</span><span class="op">)</span>  <span class="op">%&gt;%</span> <span class="co"># calculate foldchange and then </span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span><span class="op">(</span><span class="va">FoldDifference</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">0.5</span> <span class="op">&amp;</span> <span class="co"># select major fold change ASVs and then plot </span>
           <span class="va">Prevalence.C</span> <span class="op">!=</span> <span class="fl">0</span> <span class="op">&amp;</span>  <span class="co"># Remove ASVs not detected in Controls  </span>
           <span class="va">Prevalence.L2</span> <span class="op">!=</span> <span class="fl">0</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="co"># Remove ASVs not detected in L2 </span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">FoldDifference</span>, <span class="fu"><a href="https://rdrr.io/r/stats/reorder.factor.html">reorder</a></span><span class="op">(</span><span class="va">FeatureID</span> ,<span class="va">FoldDifference</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_col</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill<span class="op">=</span><span class="va">Enriched</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"ASVs"</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span> C <span class="op">=</span> <span class="st">"steelblue"</span>, L2 <span class="op">=</span> <span class="st">"brown3"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; R version 4.0.3 (2020-10-10)</span>
<span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span>
<span class="co">#&gt; Running under: Windows 10 x64 (build 19042)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Matrix products: default</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; locale:</span>
<span class="co">#&gt; [1] LC_COLLATE=English_Netherlands.1252  LC_CTYPE=English_Netherlands.1252   </span>
<span class="co">#&gt; [3] LC_MONETARY=English_Netherlands.1252 LC_NUMERIC=C                        </span>
<span class="co">#&gt; [5] LC_TIME=English_Netherlands.1252    </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attached base packages:</span>
<span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; other attached packages:</span>
<span class="co">#&gt; [1] dplyr_1.0.2       biomeUtils_0.0.09 microbiome_1.12.0 ggplot2_3.3.3    </span>
<span class="co">#&gt; [5] phyloseq_1.34.0  </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; loaded via a namespace (and not attached):</span>
<span class="co">#&gt;  [1] nlme_3.1-149        fs_1.5.0            bit64_4.0.5        </span>
<span class="co">#&gt;  [4] progress_1.2.2      rprojroot_2.0.2     tools_4.0.3        </span>
<span class="co">#&gt;  [7] bslib_0.2.4         utf8_1.2.1          R6_2.5.0           </span>
<span class="co">#&gt; [10] vegan_2.5-7         DBI_1.1.1           BiocGenerics_0.36.1</span>
<span class="co">#&gt; [13] mgcv_1.8-33         colorspace_1.4-1    permute_0.9-5      </span>
<span class="co">#&gt; [16] rhdf5filters_1.2.0  ade4_1.7-16         withr_2.4.2        </span>
<span class="co">#&gt; [19] tidyselect_1.1.0    prettyunits_1.1.1   phangorn_2.6.3     </span>
<span class="co">#&gt; [22] bit_4.0.4           compiler_4.0.3      cli_2.4.0          </span>
<span class="co">#&gt; [25] textshaping_0.3.3   Biobase_2.50.0      desc_1.3.0         </span>
<span class="co">#&gt; [28] sass_0.3.1          scales_1.1.1        quadprog_1.5-8     </span>
<span class="co">#&gt; [31] pkgdown_1.6.1       systemfonts_1.0.1   stringr_1.4.0      </span>
<span class="co">#&gt; [34] digest_0.6.27       rmarkdown_2.7       XVector_0.30.0     </span>
<span class="co">#&gt; [37] pkgconfig_2.0.3     htmltools_0.5.1.1   fastmap_1.0.1      </span>
<span class="co">#&gt; [40] rlang_0.4.10        rstudioapi_0.13     RSQLite_2.2.1      </span>
<span class="co">#&gt; [43] jquerylib_0.1.3     generics_0.1.0      jsonlite_1.7.1     </span>
<span class="co">#&gt; [46] magrittr_2.0.1      biomformat_1.18.0   Matrix_1.2-18      </span>
<span class="co">#&gt; [49] Rcpp_1.0.5          munsell_0.5.0       S4Vectors_0.28.1   </span>
<span class="co">#&gt; [52] Rhdf5lib_1.12.1     fansi_0.4.2         DECIPHER_2.18.1    </span>
<span class="co">#&gt; [55] ape_5.4-1           lifecycle_1.0.0     stringi_1.5.3      </span>
<span class="co">#&gt; [58] yaml_2.2.1          MASS_7.3-53         zlibbioc_1.36.0    </span>
<span class="co">#&gt; [61] rhdf5_2.34.0        Rtsne_0.15          plyr_1.8.6         </span>
<span class="co">#&gt; [64] grid_4.0.3          blob_1.2.1          parallel_4.0.3     </span>
<span class="co">#&gt; [67] crayon_1.4.1        lattice_0.20-41     Biostrings_2.58.0  </span>
<span class="co">#&gt; [70] splines_4.0.3       multtest_2.46.0     hms_1.0.0          </span>
<span class="co">#&gt; [73] ps_1.6.0            knitr_1.32          pillar_1.6.0       </span>
<span class="co">#&gt; [76] igraph_1.2.6        reshape2_1.4.4      codetools_0.2-16   </span>
<span class="co">#&gt; [79] stats4_4.0.3        fastmatch_1.1-0     picante_1.8.2      </span>
<span class="co">#&gt; [82] glue_1.4.2          evaluate_0.14       data.table_1.13.0  </span>
<span class="co">#&gt; [85] vctrs_0.3.4         foreach_1.5.1       gtable_0.3.0       </span>
<span class="co">#&gt; [88] purrr_0.3.4         tidyr_1.1.2         cachem_1.0.4       </span>
<span class="co">#&gt; [91] xfun_0.22           ragg_1.1.2          survival_3.2-7     </span>
<span class="co">#&gt; [94] tibble_3.0.4        iterators_1.0.13    memoise_2.0.0      </span>
<span class="co">#&gt; [97] IRanges_2.24.1      cluster_2.1.0       ellipsis_0.3.1</span></code></pre></div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Sudarshan Shetty, Person Person, Person Person.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
